cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/centrality/BetweenessCentrality.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.centrality
+import com.twitter.cassovary.graph._
+import com.twitter.cassovary.graph.tourist.PrevNbrCounter
+
+class BetweenessCentrality(graph: DirectedGraph, normalize: Boolean = true) extends AbstractCentrality(graph) {
+
+  private def combineMap [A, @specialized(Int, Double, Float, Long) B]
+  (a: Map[A,B], b: Map[A,B])(implicit ev: B => Double) : Map[A,B] = {
+    (a.toList ++ b.toList).groupBy(l => l._1).map{ t => t._1 -> t._2
+      .foldLeft(0.0) { (x,y) => x + y._2 }.asInstanceOf[B] }
+  }
+
+  def _recalc(): Unit = {
+    val tempCentralityValues = graph.foldLeft(new Array[Double](graph.maxNodeId + 1)) {
+      (partialCentralityValues, node) =>
+      val bfs = new BreadthFirstTraverser(graph, GraphDir.OutDir, Seq(node.id), Walk.Limits(), Some(new PrevNbrCounter))
+      val (nodes, depths) = bfs.foldLeft((List.empty[Node], Map.empty[Node, Int])) {
+        case ((ns, d), n) =>
+          (ns ++ List(n), d ++ Map(n -> bfs.depth(n.id).get))
+      }
+      val previous = nodes.map { n =>
```

Comment:
The calculations are a bit hard to read/understand. Can you try to explain in semi-english/pseudocode what's going on? What's the role of inboundnodes ("previous) and outboundnodes ("nearestOut") ?